### PR TITLE
Add Unified Intl.NumberFormat proposal to ECMA-402.

### PR DESCRIPTION
**Pull request by sffc**
This PR is to merge the Unified Intl.NumberFormat proposal into ECMA-402, pending final Stage 4 approval.

https://github.com/tc39/proposal-unified-intl-numberformat

### CODE REVIEW

File: spec/pluralrules.html [hunk 1]
```diff
@@ -131,7 +131,8 @@ <h1>ResolvePlural ( _pluralRules_, _n_ )</h1>
           1. Return *"other"*.
         1. Let _locale_ be _pluralRules_.[[Locale]].
         1. Let _type_ be _pluralRules_.[[Type]].
-        1. Let _s_ be ! FormatNumericToString(_pluralRules_, _n_).
+        1. Let _res_ be ! FormatNumericToString(_pluralRules_, _n_).
+        1. Let _s_ be _res_.[[FormattedString]]
```
Comment (Ms2ger) [inline spec/pluralrules.html]:
```suggestion
        1. Let _s_ be _res_.[[FormattedString]].
```

File: spec/numberformat.html [hunk 2]
```diff
@@ -452,6 +533,166 @@ <h1>UnwrapNumberFormat( _nf_ )</h1>
...
+        1. Let _notationSubPatterns_ be _dataLocaleData_.[[notationSubPatterns]].
+        1. Assert: _notationSubPatterns_ is a Record (see <emu-xref href="#sec-intl.numberformat-internal-slots"></emu-xref>).
+        1. Let _notation_ be _numberFormat_.[[Notation]].
+        1. If _notation_ is *"scientific"* or _notation_ is *"engineering"*, then
+          1. Return _notationSubPatterns_.[[scientific]].
+        1. Else if _exponent_ is not 0, then
+          1. Assert: _notation_ is *"compact"*.
+          1. Let _compactDisplay_ be _numberFormat_.[[CompactDisplay]].
+          1. Let _compactPatterns_ be _notationSubPatterns_.[[compact]].[[&gt;_compactDisplay_&lt;]].
+          1. Return _compactPatterns_.[[_exponent_]].
```
Comment (Ms2ger) [inline spec/numberformat.html]:
```suggestion
          1. Return _compactPatterns_.[[&lt;_exponent_&gt;]].
```

**Comment by sffc** [comment]
TC39 approved this for Stage 4 today, February 5, 2019.  I will address the comments here and then we should be in the clear to merge.

File: spec/numberformat.html [hunk 3]
```diff
@@ -452,6 +533,166 @@ <h1>UnwrapNumberFormat( _nf_ )</h1>
...
+        1. Let _dataLocaleData_ be _localeData_.[[&lt;_dataLocale_&gt;]].
+        1. Let _notationSubPatterns_ be _dataLocaleData_.[[notationSubPatterns]].
+        1. Assert: _notationSubPatterns_ is a Record (see <emu-xref href="#sec-intl.numberformat-internal-slots"></emu-xref>).
+        1. Let _notation_ be _numberFormat_.[[Notation]].
+        1. If _notation_ is *"scientific"* or _notation_ is *"engineering"*, then
+          1. Return _notationSubPatterns_.[[scientific]].
+        1. Else if _exponent_ is not 0, then
+          1. Assert: _notation_ is *"compact"*.
+          1. Let _compactDisplay_ be _numberFormat_.[[CompactDisplay]].
+          1. Let _compactPatterns_ be _notationSubPatterns_.[[compact]].[[&gt;_compactDisplay_&lt;]].
```
Comment (leobalter) [inline spec/numberformat.html]: typo on the &gt; and &lt; ordering

**Pull request merged**